Read pyconfig.h from ${libdir} in distutils.sysconfig

distutils.sysconfig parses pyconfig.h from ${incdir} by default, but
does not follow #include directives. This causes incorrect results in
Yocto's installation because the recipe splits pyconfig.h into
a multilib header.

Solve this by installing a copy of pyconfig.h into ${libdir}/config,
which does not need the multilib header handling and patch
distutils.sysconfig to search pyconfig.h in this folder.

Upstream-Status: Inappropriate [OE-specific]
--- a/Lib/distutils/sysconfig.py.orig	2016-08-22 08:39:00.363789250 +0200
+++ b/Lib/distutils/sysconfig.py	2016-08-22 08:39:48.017505240 +0200
@@ -233,7 +233,7 @@
         else:
             inc_dir = project_base
     else:
-        inc_dir = get_python_inc(plat_specific=1)
+        inc_dir = os.path.join(get_python_lib(plat_specific=1, standard_lib=1), "config")
     if get_python_version() < '2.2':
         config_h = 'config.h'
     else:
